[pickers] Fix `aria-labelledby` assignment to dialog (#7608)
mui/mui-x: docs/data/date-pickers/base-concepts/base-concepts.md

{{"demo": "ComponentExplorerNoSnap.js", "hideToolbar": true}}

## Accessibility

Both `Desktop` and `Mobile` Date and Time Pickers are using `role="dialog"` to display their interactive view parts and thus they should follow [Modal accessibility guidelines](/material-ui/react-modal/#accessibility).
This behavior is automated as much as possible, ensuring that the Date and Time Pickers are accessible in most cases.
A correct `aria-labelledby` value is assigned to the dialog component based on the following rules:

> - Use `toolbar` id if it is not hidden;

Reviewer: ```suggestion
- Use `toolbar` id if the toolbar is visible;
```
 or 
```suggestion
- Use the id of the toolbar if it exists;
```

> - Use `label` id (or `start` and `end` ids in case of range picker) if `toolbar` is hidden;

Reviewer: ```suggestion
- Use the id of the input label if the toolbar is hidden;
```
